webapp/src/containers/modals/SubscribeModal/filters/pipelines.tsx

```diff
@@ -0,0 +1,215 @@
+
+type PipelinesFilterProps = {
+    organization: string
+    projectId: string
+    eventType: string
+    selectedBuildPipeline: string
+    handleSelectBuildPipeline: (value: string, name?: string) => void
+    setIsFiltersError: (value: boolean) => void
+    selectedBuildStatus: string
+    handleSelectBuildStatus: (value: string, name?: string) => void
+    handleSelectReleasePipeline: (value: string, name?: string) => void
+    selectedReleasePipeline: string
+    handleSelectStageName: (value: string, name?: string) => void
+    selectedStageName: string
+    handleSelectApprovalType: (value: string, name?: string) => void
+    selectedApprovalType: string
+    handleSelectApprovalStatus: (value: string, name?: string) => void
+    selectedApprovalStatus: string
+    selectedReleaseStatus: string
+    handleSelectReleaseStatus: (value: string, name?: string) => void
+}
+
+const PipelinesFilter = ({
+    organization,
+    projectId,
+    eventType,
+    selectedBuildPipeline,
+    handleSelectBuildPipeline,
+    setIsFiltersError,
+    selectedBuildStatus,
+    handleSelectBuildStatus,
+    handleSelectReleasePipeline,
+    selectedReleasePipeline,
+    handleSelectStageName,
+    selectedStageName,
+    selectedApprovalType,
+    handleSelectApprovalType,
+    handleSelectApprovalStatus,
+    selectedApprovalStatus,
+    selectedReleaseStatus,
+    handleSelectReleaseStatus,
+}: PipelinesFilterProps) => {
+    const {buildStatusOptions, releaseApprovalTypeOptions, releaseApprovalStatusOptions, releaseStatusOptions, subscriptionFiltersForPipelines, subscriptionFiltersNameForPipelines} = pluginConstants.form;
+
+    const {
+        getApiState,
+        makeApiRequestWithCompletionStatus,
+    } = usePluginApi();
+
+    const getSubscriptionFiltersRequest = useMemo<GetSubscriptionFiltersRequest>(() => ({
+        organization,
+        projectId,
+        filters: subscriptionFiltersForPipelines,
+        eventType,
+        releasePipelineId: selectedReleasePipeline,
+    }), [organization, projectId, eventType, subscriptionFiltersForPipelines, selectedBuildPipeline, selectedReleasePipeline]);
+
+    useEffect(() => {
+        if (eventType) {
+            makeApiRequestWithCompletionStatus(
+                pluginConstants.pluginApiServiceConfigs.getSubscriptionFilters.apiServiceName,
+                getSubscriptionFiltersRequest,
+            );
+        }
+    }, [getSubscriptionFiltersRequest]);
+
+    const {data, isLoading, isError, isSuccess} = getApiState(
+        pluginConstants.pluginApiServiceConfigs.getSubscriptionFilters.apiServiceName,
+        getSubscriptionFiltersRequest as APIRequestPayload,
+    );
+    const filtersData = data as GetSubscriptionFiltersResponse || [];
+
+    useEffect(() => {
+        if (isError && !isSuccess) {
+            setIsFiltersError(true);
+        } else {
+            setIsFiltersError(false);
+        }
+    }, [isLoading, isError, isSuccess]);
```

Comment:
is `isLoading` required here?

Reply:
I think it is used to check for error when loading is completed. Right @avas27JTG ?

Reply:
on completion of loading, either the `isError` or `isSuccess` will change. So, I don't think we require that

Reply:
yes we can remove `isLoading`